components/builder-depot-client/src/lib.rs

```diff
@@ -1189,9 +1176,7 @@ fn package_search(term: &str) -> String {
     format!("depot/pkgs/search/{}", encoded_term)
 }
 
-fn channel_package_path<I>(channel: &str, package: &I) -> String
-where
-    I: Identifiable,
+fn channel_package_path(channel: &str, package: &PackageIdent) -> String
```

Comment:
We really need to extract the `Channel` type we've got in the `common` crate and start using it more. String-based APIs make me sad 😦 

(Not work for this PR, of course... I'm just musing out loud.)

Reply:
Oh man YES, I would love to get all of these string types and interstitial conversion things swapped out for real data types.

Reply:
I'm seeing `String`s everywhere that probably should have been types. Especially in `Option<String>` and `Option<&str>` cases. Slow and steady progress and we may be able to clean things up here--at no extra runtime cost either, purely developer safety and slightly more expressive code.

Reply:
Most certainly. I considered actually correcting these but I opted to skip it for now with the intention of cleaning them up later.

Reply:
Totally! Maybe we need a special Hack Day(s) (2 max?) dedicated to De-Stringing Habitat where easy and possible. Low hanging fruit and all.

Reply:
Yeah I'm thinking during our ABSOLUTELY REAL AND NOT COMPLETELY MADE UP team offsite in Banff, what say you?

Reply:
🇨🇦 !!

Reply:
Yes, the inter-season (post-camping and pre-skiing) is almost upon us!